lib/fog/compute/google/requests/reset_windows_password.rb

```diff
@@ -0,0 +1,143 @@
+# Copyright 2015 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Changes:
+# March 2020: Modified example found here:
+# https://github.com/GoogleCloudPlatform/compute-image-windows/blob/master/examples/windows_auth_python_sample.py
+# to enable fog-google to change windows passwords.
+
+require "openssl"
+require "base64"
+require "json"
+
+module Fog
+  module Compute
+    class Google
+      class Mock
+        def reset_windows_password(_instance, _zone, _users)
+          Fog::Mock.not_implemented
+        end
+      end
+
+      class Real
+        ##
+        # Resets Windows passwords for users on Google's Windows based images.  Code based on Google provided example.
+        #
+        # @param instance [String] the name of the instance
+        # @param zone [String] the name of the zone of the instance
+        # @param user [String] the user whose password should be reset
+        #
+        # @return [String] new password
+        #
+        # @see https://cloud.google.com/compute/docs/instances/windows/automate-pw-generation
+        def reset_windows_password(instance, zone, user)
```

Comment:
Is it possible to add a test for this? I know it's additional effort on your part, but I'm happy to help make this as painless as possible.

I think it's important since I'm not super-savvy with Windows on GCP and I think having a test will make sure we don't break your critical paths going forward.

It should be straightforward when the method is in the model, you can use this test as a template: https://github.com/fog/fog-google/blob/a82c2c598ca19edf58b3087cc4cf7266d733da8f/test/integration/compute/core_compute/test_servers.rb#L98-L106

The hash passed to `@factory.create` just means adding or overriding the server options created in the factory here:
https://github.com/fog/fog-google/blob/d9fafff7fe1349e7bf1b3b747b4fa4f450697f1c/test/integration/factories/servers_factory.rb#L24-L30

You'll just need to instantiate your own disk factory and then override the image here:
https://github.com/fog/fog-google/blob/d9fafff7fe1349e7bf1b3b747b4fa4f450697f1c/test/integration/factories/disks_factory.rb#L19

Reply:
I added a test to `test/integration/core_compute/test_servers.rb`.  I am not sure if I did it exactly as you specified because I used the existing DiskFactory.  One issue I ran into when using both my own instance and the `disks` instance was that the Server factory will create a disk even if you override the `:disks` parameter in the `@factory.create` call.  This would cause both disks to be named the same thing and error out.  So I just tack on a "-2" and use the existing DiskFactory.

Also, as the test required checking the serial port output, I went ahead and modified the `server` model to allow more than just serial port 1 to be read, though it will default to 1 so that existing functionality should be preserved.

Reply:
It's close enough to what's needed and I'll just make a small fix later. Thank you so much! :)